powertools-parameters/src/main/java/software/amazon/lambda/powertools/parameters/exception/ProviderException.java

```diff
@@ -0,0 +1,12 @@
+package software.amazon.lambda.powertools.parameters.exception;
+
+public class ProviderException extends RuntimeException {
```

Comment:
May be add some java docs to this exception ? I think since this is an exceptional scenario which would probably indicate some bug etc? May be just throw RuntimeException or IllegalStateException or something ? If at all we decide to have a custom exception, please add some java docs